internal/backend/backend_retry.go

```diff
@@ -32,8 +33,21 @@ func NewRetryBackend(be restic.Backend, maxTries int, report func(string, error,
 	}
 }
 
+func isRecoverable(err error) bool {
+	return !strings.HasSuffix(err.Error(), "no space left on device")
```

Comment:
I think this is a good idea in general, but I'm not sure the check belongs here, rather than in the individual backends. I know that would tie knowledge of the retrying logic into the backends, but there are many conditions that are not recoverable (dropped connections being the foremost) and enumerating them here would become messy. It would also allow a more reliable check than formatting the error and checking for string patterns.

Reply:
I agree. The backend should wrap the error (e.g. within a struct) so that the returned error has a new method (e.g. `IsFatal()` or so), so we can test that in the retry logic. Unfortunately, it's not so simple :)

Reply:
The backoff package actually has an error type for this (backoff.PermanentError), but I found that installing that type is a non-trivial undertaking.

Reply:
Used Fatal error to break cycle instead of custom function. Errors in mkdir and io.Copy are considered fatal.